src/battle_game/core/unit.h

```diff
@@ -42,6 +45,13 @@ class Unit : public Object {
     health_ = std::clamp(new_health, 0.0f, 1.0f);
   }
 
+  virtual void RenderLifeBar() {
+    if (!lifebar_) {
+      lifebar_ = new LifeBar(game_core_, id_);
```

Comment:
We do not suggest using “new” in a formal project, could be unsafe.
Use smart pointers instead.